features/steps/jobs_feature.go

```diff
@@ -0,0 +1,227 @@
+		URI:             mongoFeature.Server.URI(),
+	}
+	ctx := context.Background()
+	if dbErr := mongodb.Init(ctx, cfg); dbErr != nil {
+		return nil, fmt.Errorf("failed to initialise mongo DB: %w", dbErr)
+	}
+
+	f.MongoClient = mongodb
+
+	f.AuthFeature = authFeature
+	cfg.ZebedeeURL = f.AuthFeature.FakeAuthService.ResolveURL("")
+
+	f.SearchFeature = searchFeature
+	cfg.SearchAPIURL = f.SearchFeature.FakeSearchAPI.ResolveURL("")
+
+	messageProducer := kafkatest.NewMessageProducer(true)
+	messageProducer.CheckerFunc = DoGetKafkaProducerChecker
+	f.MessageProducer = messageProducer
+
+	kafkaProducer := &event.ReindexRequestedProducer{
+		Marshaller: schema.ReindexRequestedEvent,
+		Producer:   messageProducer,
+	}
+	f.KafkaProducer = kafkaProducer
+
+	f.kafkaProducerOutputData = make(chan []byte, 9999)
+	f.quitReadingOutput = make(chan bool)
+	f.readOutputMessages()
+
+	err = runJobsFeatureService(ctx, f, cfg, svcErrors)
+	if err != nil {
+		return nil, fmt.Errorf("failed to run JobsFeature service: %w", err)
+	}
+
+	return f, nil
+}
+
+// runJobsFeatureService uses the InitialiserMock to create the mock services that are required for testing the JobsFeature
+// It then uses these to create the external serviceList, which it passes into the service.Run function along with the fake identity and search clients.
+func runJobsFeatureService(ctx context.Context, f *JobsFeature, cfg *config.Config, svcErrors chan error) error {
+	var err error
+	initFunctions := &serviceMock.InitialiserMock{
+		DoGetHealthCheckFunc:           f.DoGetHealthcheckOk,
+		DoGetHTTPServerFunc:            f.DoGetHTTPServer,
+		DoGetMongoDBFunc:               f.DoGetMongoDB,
+		DoGetAuthorisationHandlersFunc: f.DoGetAuthorisationHandlers,
+		DoGetKafkaProducerFunc:         f.DoGetKafkaProducer,
+	}
+
+	serviceList := service.NewServiceList(initFunctions)
+	testIdentityClient := clientsidentity.New(cfg.ZebedeeURL)
+	testSearchClient := clientssitesearch.NewClient(cfg.SearchAPIURL)
+
+	f.svc, err = service.Run(ctx, cfg, serviceList, "1", "", "", svcErrors, testIdentityClient, taskNames, testSearchClient)
+	return err
+}
+
+// InitAPIFeature initialises the APIFeature that's contained within a specific JobsFeature.
+func (f *JobsFeature) InitAPIFeature() *componentTest.APIFeature {
+	f.APIFeature = componentTest.NewAPIFeature(f.InitialiseService)
+
+	return f.APIFeature
+}
+
+// Reset sets the resources within a specific JobsFeature back to their default values.
+func (f *JobsFeature) Reset(mongoFail bool) error {
+	if mongoFail {
+		f.MongoClient.Database = "lost database connection"
+	} else {
+		f.MongoClient.Database = utils.RandomDatabase()
+	}
+	if f.Config == nil {
+		cfg, err := config.Get()
+		if err != nil {
+			return fmt.Errorf("failed to get config: %w", err)
+		}
+		f.Config = cfg
+	}
+
+	return nil
+}
+
+// Close stops the *service.Service, which is pointed to from within the specific JobsFeature, from running.
+func (f *JobsFeature) Close() error {
```

Comment:
Does is make sense to call this JobsFeature? Is it not SearchReindexAPIFeature?

Reply:
Done - will be updated once the new version of dp-net is available